src/skills/index.js

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash'
+
+import { validateFlowSchema } from '../dialog/validator'
+
+const SKILLS_PREFIX = 'botpress-skill-'
+const SKILLS_PREFIX_REGEX = new RegExp('^' + SKILLS_PREFIX)
+
+export default class SkillsManager {
+  constructor({ logger }) {
+    this.logger = logger
+
+    this._log('info', `[Skills] Initiated`)
+  }
+
+  _log = (level, msg) => {
+    this.logger && this.logger[level] && this.logger[level](msg)
+  }
+
+  registerSkillsFromModules = modules => {
+    this._skills = modules.filter(mod => SKILLS_PREFIX_REGEX.test(mod.name)).reduce((acc, curr) => {
+      if (!curr.handlers.generate) {
+        this._log('warn', `Skill "${curr.name}" has no flow generator ("generate" method exposed)`)
+        return acc
+      }
+
+      if (!_.isFunction(curr.handlers.generate)) {
+        this._log('warn', `Skill "${curr.name}" generator is not a valid function`)
+        return acc
+      }
+
+      acc[curr.name] = curr.handlers.generate
+      return acc
+    }, {})
+
+    this._log('info', `[Skills] Loaded ${_.keys(this._skills).length} skills`)
+  }
+
+  generateFlow = async (skillId, data) => {
```

Comment:
This doesn't seem to be called unbound so can be a normal method
```
async generateFlow(skillId, data) { ... }
```

My suggestion to use fat arrows where possible primarily applies to stand-alone functions. For class methods normal syntax is better because it puts the method on the prototype (class instance props on contrary create methods copies and attach them directly to `this`)

Reply:
Same for couple of other methods. Prop instances are useful when used unbound, like passing as callbacks to react components, or chained in promises like `.then(this.someMethod)`